core/controllers/voice_artist.py

```diff
@@ -38,15 +39,49 @@ class AudioUploadHandler(base.BaseHandler):
     to the local datastore in dev).
     """
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'raw_audio_file': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                    'id': 'is_nonempty'
+                    }]
+                }
+            },
+            'filename': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': r'^[A-Za-z0-9-]+[.][a-z0-9]+$'
+                    }]
```

Comment:
You should probably check for allowed extensions as the code below does, then you can remove the code below.

Reply:
@vojtechjelinek Do I need to create a new validator function in 'schema_utils.py' to check if the extension is from accepted list of formats?

Reply:
Yep